ipa-core/src/protocol/context/dzkp_field.rs

```diff
@@ -57,6 +55,16 @@ pub trait DZKPBaseField: PrimeField {
     ) -> Vec<UVSingleBlock<Self>>;
 }
 
+impl FromIterator<Fp61BitPrime> for [Fp61BitPrime; B] {
+    fn from_iter<T: IntoIterator<Item = Fp61BitPrime>>(iter: T) -> Self {
+        let mut out = [Fp61BitPrime::ZERO; B];
+        for (i, elem) in iter.into_iter().take(B).enumerate() {
```

Comment:
I think it makes sense to panic here instead of taking B many. We don't want to drop elements from the iterator accidentally.